sympy/diffgeom/diffgeom.py

```diff
@@ -120,134 +157,155 @@ def dim(self):
+    def __new__(cls, name, patch, symbols=None, relations={}, **kwargs):
         if not isinstance(name, Str):
             name = Str(name)
-        # names is not in args because it is related only to printing, not to
-        # identifying the CoordSystem instance.
-        obj = super().__new__(cls, name, patch)
-        if not names:
-            names = ['%s_%d' % (name, i) for i in range(patch.dim)]
-        obj._names = tuple(str(i) for i in names)
-        obj.patch.coord_systems.append(obj)
-        obj.transforms = {}
-        # All the coordinate transformation logic is in this dictionary in the
-        # form of:
-        #  key = other coordinate system
-        #  value = tuple of  # TODO make these Lambda instances
-        #          - list of `Dummy` coordinates in this coordinate system
-        #          - list of expressions as a function of the Dummies giving
-        #          the coordinates in another coordinate system
-        obj._dummies = [Dummy(str(n)) for n in names]
+
+        # canonicallize the symbols
+        if symbols is None or any(not isinstance(s, Symbol) for s in symbols) :
+            # support deprecated signature
+            SymPyDeprecationWarning(
+                feature="Class signature 'names' of CoordSystem",
+                useinstead="class signature 'symbols'",
+                issue=19321,
+                deprecated_since_version="1.7"
+            ).warn()
+
+            if symbols is None:
+                names = kwargs.get('names', None)
+            else:
+                names = symbols
+
+            if names is None:
+                symbols = Tuple(*[Symbol('%s_%d' % (name, i), real=True) for i in range(patch.dim)])
+            else:
+                symbols = Tuple(*[Symbol(n, real=True) for n in names])
+        else:
+            symbols = Tuple(*[Symbol(s.name, **s._assumptions.generator) for s in symbols])
+
+        # canonicallize the relations
+        rel_temp = {}
+        for k,v in relations.items():
+            s1, s2 = k
+            if not isinstance(s1, Str):
+                s1 = Str(s1)
+            if not isinstance(s2, Str):
+                s2 = Str(s2)
+            key = Tuple(s1, s2)
+            rel_temp[key] = v
+        relations = Dict(rel_temp)
+
+        # construct the object
+        obj = super().__new__(cls, name, patch, symbols, relations)
+
+        # Add deprecated attributes
+        obj.transforms = _deprecated_dict(
+            "Mutable CoordSystem.transforms",
+            "'relations' parameter in class signature",
+            19321,
+            "1.7",
+            {}
+        )
+        obj._names =_deprecated_list(
+                "CoordSystem._names",
+                "CoordSystem.symbols",
+                19321,
+                "1.7",
+                [str(n) for n in symbols]
+            )
+        obj.patch.coord_systems.append(obj) # deprecated
+        obj._dummies = _deprecated_list(
```

Comment:
These are private methods, so it can be overwhelming to raise deprecation warnings for `_names` and `_dummies`

Reply:
I guess you're right. I un-deprecated them.